README.md

```diff
@@ -96,6 +96,32 @@ const orderId = await cowSdk.cowApi.sendOrder({
 console.log(`https://explorer.cow.fi/rinkeby/orders/${orderId}`)
 ```
 
+SDK also allows you to decode the appData hash to get the ipfs appData document
+
+```js
+  const hash = '0xa6c81f4ca727252a05b108f1742a07430f28d474d2a3492d8f325746824d22e5'
+  const appDataDoc  = await decodeAppData(hash)
+  console.log(appDataDoc) 
+  /* {
+      "appCode": "CowSwap",
+      "metadata": {
+          "referrer": {
+              "address": "0x1f5B740436Fc5935622e92aa3b46818906F416E9",
+              "version": "0.1.0"
+          }
+      },
+      "version": "0.1.0"
+  } *.
+```
+
+Or decode an IPFS mutlihash
+
+```js
+  const hash = 'QmUf2TrpSANVXdgcYfAAACe6kg551cY3rAemB7xfEMjYvs'
+  const decodedMultihash  = await decodeMultihash(hash)
+  console.log(decodedMultihash) // 0x5ddb2c8207c10b96fac92cb934ef9ba004bc007a073c9e5b13edc422f209ed80
+```
+
```

Comment:
Thanks for this, much better to understand how to use the lib with this :)

And now with more visibility I am wondering:
1. The name of this function. It's going from CID to appData hash, so `decodeMultihash` does not really tell me much about what I can use it for
2. The need for this function. Nowhere in the process we expose the CID associated with an appData.
  Maybe we should have an intermediate/additional fn to convert from appData to CID without loading the associated data? 
  Maybe we should have an intermediate/additiona fn to load IPFS data from CID?
  Not sure, that would be kind of replicating an IPFS API.
  Or maybe it'll make more sense once we have an "appData creation api" where users can edit their fields and upload it to IPFS getting back a CID/appData hash
  
What do you guys think about this? @matextrem @ramirotw @anxolin

Reply:
Agree, great example! Good job 

1. True, maybe name can be revisited. I'm not sure how to properly give a name to this, since is such a weird thing :) `cidToMultihashDigestHex` ? or cidToHex` if we wan to abuse the language to make it simpler. Or maybe `cidToAppDataHex`. 
2. 
> Maybe we should have an intermediate/additional fn to convert from appData to CID without loading the associated data?

Agree, just if we opt for `cidToAppDataHex` before, we can do the  `appDataHexToCid`. 

> Maybe we should have an intermediate/additiona fn to load IPFS data from CID?
Since we have the implementation, i find it ok to expose it to give options to SDK users. 

> Or maybe it'll make more sense once we have an "appData creation api" where users can edit their fields and upload it to IPFS getting back a CID/appData hash

Not sure if I get you. But if you meant to add appDataApi here: 
<img width="451" alt="image" src="https://user-images.githubusercontent.com/2352112/157773595-9377ca9d-1756-4378-9348-0d6deee09594.png">

I think is good. The only think i'd like to make sure, is that people don't end up pulling unnecesary depedencies if they don't use one API.

Reply:
If we want to go the api route, we can have:

`sdk.cowApi`
`sdk.signingApi`
`sdk.appDataApi` --> I hate the name. This is how its called the field in the order. Other name could be `sdk.metadataApi` because allows to generate arbitrary metadata and attach it to an order. Not strong opinion, just thinking aloud

Reply:
Yes, `sdk.metadataApi` is part of what I meant.

The other part is to have the metadata creation/upload as well.

Such as providing the metadata schema/validating it and storing it to IPFS.

Imagine someone wants to create a custom UI and have all orders placed through it clearly identified:

```json
  {
      "appCode": "My nice UI",
      "metadata": {
          "referrer": {
              "address": "0x1f5B740436Fc5935622e92aa3b46818906F416E9",
              "version": "0.1.0"
          }
      },
      "version": "0.1.0"
  }
```

The API would expose the methods for creating/validating the JSON above.
And, if the user wants, upload to IPFS on their behalf and return the associated CID and/or appDataHash